src/app/clusters/im-compatibility/ember-compatibility-functions.cpp

```diff
@@ -0,0 +1,89 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *          Contains the functions for compatibility with ember ZCL inner state
+ *          when calling ember callbacks.
+ */
+
+#include "ember-compatibility-functions.h"
+
+#include <lib/core/CHIPCore.h>
+#include <lib/core/CHIPTLV.h>
+#include <lib/support/CodeUtils.h>
+#include <util/util.h>
+
+namespace chip {
+namespace app {
+namespace Compatibility {
+namespace {
+EmberAfClusterCommand imCompatibilityEmberAfCluster;
+EmberApsFrame imCompatibilityEmberApsFrame;
+EmberAfInterpanHeader imCompatibilityInterpanHeader;
+Command * currentCommandObject;
+} // namespace
+
+void SetupEmberAfObjects(Command * command, ClusterId clusterId, CommandId commandId, EndpointId endpointId)
+{
+    imCompatibilityEmberApsFrame.clusterId           = clusterId;
+    imCompatibilityEmberApsFrame.destinationEndpoint = endpointId;
+    imCompatibilityEmberApsFrame.sourceEndpoint      = endpointId;
```

Comment:
I think `sourceEndpoint` is basically `1` for now. You may want to use that instead of `endpointId`. For applications with multiple endpoints it will send the response on the wrong client endpoint otherwise.